File: luxview-engine/internal/model/community.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type CommunityPost struct {
	ID          uuid.UUID `json:"id"`
	AppID       uuid.UUID `json:"app_id"`
	Game        string    `json:"game"`
	GameName    string    `json:"game_name"`
	DisplayName string    `json:"display_name"`
	Title       string    `json:"title"`
	Body        string    `json:"body"`
	CreatedAt   time.Time `json:"created_at"`
}

type CommunityMessage struct {
	ID        uuid.UUID `json:"id"`
	Username  string    `json:"username"`
	Body      string    `json:"body"`
	CreatedAt time.Time `json:"created_at"`
}

type CommunityGamePlayers struct {
	AppID       string `json:"app_id"`
	Game        string `json:"game"`
	Name        string `json:"name"`
	DisplayName string `json:"display_name"`
	Players     int    `json:"players"`
	MaxPlayers  int    `json:"max_players"`
}

type CommunitySnapshot struct {
	PlayersOnline int                    `json:"players_online"`
	ChatOnline    int                    `json:"chat_online"`
	Games         []CommunityGamePlayers `json:"games"`
	Posts         []CommunityPost        `json:"posts"`
	Chat          []CommunityMessage     `json:"chat"`
}

File: luxview-dashboard/src/api/community.ts
import { api } from './client';

export interface CommunityPost {
  id: string;
  appId: string;
  game: string;
  gameName: string;
  displayName: string;
  title: string;
  body: string;
  createdAt: string;
}

export const communityApi = {
  async listPosts(appId: string): Promise<CommunityPost[]> {
    const { data } = await api.get<CommunityPost[]>(`/apps/${appId}/community/posts`);
    return data ?? [];
  },

  async createPost(appId: string, title: string, body: string): Promise<CommunityPost> {
    const { data } = await api.post<CommunityPost>(`/apps/${appId}/community/posts`, { title, body });
    return data;
  },

  async deletePost(appId: string, postId: string): Promise<void> {
    await api.delete(`/apps/${appId}/community/posts/${postId}`);
  },
};
